Feat/collection novidades

ThaisMap opened the pull request:
Adiciona entidade, repository e services de Novidades
Adiciona páginas para CRUD de Novidades
Ajusta MenuListItems pra redirecionar para páginas
Adiciona componente SelectionField
Adiciona alerta para novidades

Closes #178378600

ThaisMap committed:
feat: adiciona entidade, repository e services de Novidades

ThaisMap committed:
feat: adiciona entidade, repository e services de Novidades

ThaisMap committed:
feat: altera campos no repository

- remove opcional de alguns atributos
- remove atributo slug (usar o id no lugar)
- corrige erros no NovidadesRepository

ThaisMap committed:
feat: altera campos no repository

- remove opcional de alguns atributos
- remove atributo slug (usar o id no lugar)
- corrige erros no NovidadesRepository

ThaisMap committed:
feat: CRUD Novidades

- Adiciona link para Novidades no menu lateral
- Adiciona componente NovidadesForm
- Adiciona componente SelectionField
- Adiciona páginas de index, cadastro e edição de novidades
- Adiciona lista de páginas do app

ThaisMap committed:
feat: CRUD Novidades

- Adiciona link para Novidades no menu lateral
- Adiciona componente NovidadesForm
- Adiciona componente SelectionField
- Adiciona páginas de index, cadastro e edição de novidades
- Adiciona lista de páginas do app

ThaisMap committed:
Merge remote-tracking branch 'origin/feat/collection-novidades' into feat/collection-novidades

ThaisMap committed:
fix: integra alterações do AdminBase

- Tira o componente diretamente das páginas

ThaisMap committed:
feat: Adiciona alert para novidades

ThaisMap committed:
fix: ajusta alerta

- Corrige erro que não deixava alertas aparecer
- Tira console.logs

kellynvd commented on src/components/admin/NovidadesForm.tsx:
```diff
@@ -0,0 +1,244 @@
+  )
+  const [dataFinal, setDataFinal] = useState(novidade?.dataFinal || new Date())
+  const [permanencia, setPermanencia] = useState(
+    novidade?.autoDispensar ? 'uma vez' : 'sempre' || 'sempre'
```
Acho que esse segundo 'sempre' não precisa né?

ThaisMap committed:
style: Ajusta layout

- Adiciona erro no componente dateInput
- Adiciona opções no componente selectionField
- Altera ícone de novidades no menu lateral
- Usa um enum para opções de permanência da novidade
- Usa radiogroup customizado no form de novidades

ThaisMap committed:
fix: ajusta tipagem no SelectionField

kellynvd commented on src/pages/admin/novidades/nova.tsx:
```diff
@@ -0,0 +1,29 @@
+  return (
+    <Container>
+      <Box className={classes.tituloContainer}>
+        <Typography className={classes.titulo}>Incluir Novidade</Typography>
```
```suggestion
        <Typography className={classes.titulo}>Nova novidade</Typography>
```

kellynvd committed:
Update src/pages/admin/novidades/nova.tsx

kellynvd reviewed (approved):
:rocket:
